### PR TITLE
Apply limit in WebUI when query state or error type filters update

### LINKED ISSUE #8185: Trino/Presto UI does not respect show limit when query state filter is changed
When you change a query state filter e.g. showing 'Finished' queries or 'User Error' queries, the Show Limit is not respected immediately resulting in all queries being rendered. If a re-order interval is set then Show Limit is applied on the next query refresh. If there are a large number of queries in the history this can result in the browser crashing on the first render.

The following two calls to sortAndLimitQueries looks like they are missing a last parameter of this.state.maxQueries resulting in no limit being applied

https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx#L512
https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx#L544

### PR DESCRIPTION
**Pull request by cpsnowden**
Resolves #8185

### CODE REVIEW

**Comment by Praveen2112** [comment]
Thanks for working on this. Is it possible to share a screenshot of the UI before and after this change ?

**Comment by cpsnowden** [comment]
@Praveen2112 yeah - will get some screenshots of the UI pre/post showing the diff in behaviour.

Also have emailed the I-CLA.

**Comment by cpsnowden** [comment]
Attached shows the behaviour change.

1. Ran 21 queries
2. Selected Show Limit 20
3. Click Show Finished

Before change the UI shows 21 queries (and doesn't apply the limit)
[Trino UI - Before Change.pdf](https://github.com/trinodb/trino/files/6600224/Trino.UI.-.Before.Change.pdf)

After change the UI correctly restricts to 20 queries
[Trino UI - After Change.pdf](https://github.com/trinodb/trino/files/6600223/Trino.UI.-.After.Change.pdf)

File: core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx [hunk 1]
```diff
@@ -509,7 +509,7 @@ export class QueryList extends React.Component {
         }
 
         const filteredQueries = this.filterQueries(this.state.allQueries, newFilters, this.state.errorTypeFilters, this.state.searchString);
-        this.sortAndLimitQueries(filteredQueries, this.state.currentSortType, this.state.currentSortOrder);
+        this.sortAndLimitQueries(filteredQueries, this.state.currentSortType, this.state.currentSortOrder, this.state.maxQueries);
```
Comment (findepi) [inline core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx:512]: We should also remove the `!== 0` part of https://github.com/trinodb/trino/blob/30b6b68f6c2f98811684faca8237698d890dddbc/core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx#L274
it becomes redundant now

Comment (cpsnowden) [reply core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx:512]: Is it not a performance optimisation for the following two cases - to remove a redundant splice?

https://github.com/trinodb/trino/blob/7989be1420c7901059ee908e91c97d283195da7a/core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx#L362

and

https://github.com/trinodb/trino/blob/7989be1420c7901059ee908e91c97d283195da7a/core/trino-main/src/main/resources/webapp/src/components/QueryList.jsx#L366

**Comment by cpsnowden** [comment]
@cla-bot check

**Comment by findepi** [comment]
@raghavsethi can you please take a look?

**Comment by cpsnowden** [comment]
@raghavsethi any thoughts on this change?

**Comment by raghavsethi** [comment]
Change looks good to me, but my personal laptop is borked rn, so I haven't been able to verify the change locally. @findepi if you could do that, then we're good to go.

**Comment by findepi** [comment]
thanks @raghavsethi 

@dedep can you PTAL as well?

**Comment by cpsnowden** [comment]
@dedep are you able to take a look?

**Comment by cpsnowden** [comment]
@dedep @findepi @raghavsethi @electrum @aalbu is there any feedback pending on this PR?

**Comment by cpsnowden** [comment]
@findepi - is anyone able to take a quick look at this? 

There are workarounds to stop the UI crashing when there are a lot of queries (if you know them) - but keen to get a permanent fix in :)

**Review by martint — APPROVED** [review]

**Pull request merged**